pages/course/[course]/assignment/[assignment]/submit/index.tsx

```diff
@@ -70,15 +71,32 @@ const AssignmentUpload: NextPage<AssignmentProps> = ({ id, courseId, assignment
   const user = useAppSelector((store) => store.user);
   const supabase = useSupabaseClient();
   const [userId, setUserId] = useState<number | undefined>(undefined);
-  
-  
+  const [lang, setLang] = useState<string>("");
+
   useEffect(() => {
+    const setLanguage = () => {
+      switch (assignment.language) {
+        case "c/c++": {
+          return ".cpp, .c";
+        }
+        case "python": {
+          return ".py";
+        }
+        case "javascript": {
+          return ".js";
+        }
+        case "java": {
+          return ".java";
+        }
+      }
+      return ".zip"
+    };
+
```

Comment:
I don't think language needs to be state, also don't think it needs to be in useEffect because we already know what it is when the page loads.

Reply:
would setLanguage make more sense as a map? You can use nullish coalescence to safely fallback to .zip. We probably want to at least rename the function though, because setLanguage doesn't actually set anything - maybe langToFileExtension?

```ts
> const myMap = new Map([
...   [1, "one"],
...   [2, "two"],
...   [3, "three"],
... ]);
undefined
> myMap.get(1) ?? "unknown";
'one'
> myMap.get(6) ?? "unknown";
'unknown'
```

Reply:
we're also using `lang` and `language` in the same scope, which is a bit confusing. We could rename `lang` to `fileExtension` or something similar since that's a more accurate reflection of what that variable holds.